app/models/piece.rb

```diff
@@ -51,18 +51,66 @@ def check_vertical_obstructions(x, y)
     return false
   end
 
-  # check if a diagnol move is blocked
+  # moved this call into a separate method for DRY reasons
+  def check_temporary_values(x_temp, y_temp)
+    game.piece_in_space?(x_temp, y_temp)
+  end
+
+  # check if a diagonal move is blocked
   def check_diagonal_obstructions(x, y)
-    # for each value x in the x_range *combined with* with each value y from the y_range, check
-    # if there is a piece in the corresponding space
-    x_range(x).each do |x|
-      y_range(y).each do |y|
-        if game.piece_in_space?(x, y)
+    # if desired move is diagonal up and to the right
+    if x > position_x && y > position_y
+      x_temp = position_x + 1
+      y_temp = position_y + 1
+      while x_temp <= x && y_temp <= y
+        if check_temporary_values(x_temp, y_temp)
           return true
+        else
+          return false
         end
+        x_temp += 1
+        y_temp += 1
+      end
+    # if desired move is diagonal up and to the left
+    elsif x < position_x && y > position_y
+      x_temp = position_x - 1
+      y_temp = position_y + 1
+      while x_temp >= x && y_temp <= y
+        if check_temporary_values(x_temp, y_temp)
+          return true
+        else
+          return false
+        end
+        x_temp -= 1
+        y_temp += 1
+      end
+    # if desired move is diagonal down and to the right
+    elsif x > position_x && y < position_y
+      x_temp = position_x + 1
+      y_temp = position_y - 1
+      while x_temp <= x && y_temp >= y
+        if check_temporary_values(x_temp, y_temp)
+          return true
+        else
+          return false
+        end
+        x_temp += 1
+        y_temp -= 1
+      end
+    # if desired move is diagonal down to the left
+    elsif x < position_x && y < position_y
+      x_temp = position_x - 1
+      y_temp = position_y - 1
+      while x_temp >= x && y_temp >= y
+        if check_temporary_values(x_temp, y_temp)
+          return true
+        else
+          return false
+        end
+        x_temp += 1
+        y_temp -= 1
```

Comment:
So I created a test suite to run this through in rspec and it seems that not all cases of checking for diagonal obstructions are working. The example I used was 
`piece_1.position_x = 1`
`piece_1.position_y = 1`
`piece_2.position_x = 5`
`piece_2.position_y = 5`

Then the test was run like
`piece_1.is_obstructed?(6, 6)`

The test returned false when it should return true. 

I haven't had a chance to really analyze all of your code but this seems to be working better than the previous version. Considering we have a lot of other things to do, I think we should merge it and continue to improve it as we move along.